rosbag2_storage_default_plugins/include/rosbag2_storage_default_plugins/sqlite/sqlite_storage.hpp

```diff
@@ -70,6 +71,11 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
 
   std::string get_storage_identifier() const override;
 
+  // APIs to handle/control transactions
+  void activate_transaction();
+
+  void commit_transaction();
+
```

Comment:
Do these functions have to be public?

Reply:
Also, I think these two functions make more sense in the `SQLiteWrapper` class as they are really just implementation details and don't need any visibility in the storage implementation. 
That would also resolve the need for the `get_db_handle()` function.

Reply:
Hi Karsten, the problem is the logic of keepign count of the number of inserts is within sqlite_storage.cpp, and to move that around looked like too much changes for me. But I'll still consider your comment.